src/rpcnet.cpp

```diff
@@ -165,6 +167,114 @@ Value getpeerinfo(const Array& params, bool fHelp)
     return ret;
 }
 
+Value gettrafficshaping(const Array& params, bool fHelp)
+{
+    string strCommand;
+    if (params.size() == 1)
+      {
+        strCommand = params[0].get_str();
+      }
+
+    if (fHelp || (params.size() != 0) )     
+        throw runtime_error(
+            "gettrafficshaping"
+            "\nReturns the current settings for the network send and receive bandwidth and burst in kilobytes per second.\n"
+            "\nArguments: None\n"
+            "\nResult:\n"
+            "  {\n"
+            "    \"sendBurst\" : 40,   (string) The maximum send bandwidth in Kbytes/sec\n"
+            "    \"sendAve\" : 30,   (string) The average send bandwidth in Kbytes/sec\n"
+            "    \"recvBurst\" : 20,   (string) The maximum receive bandwidth in Kbytes/sec\n"
+            "    \"recvAve\" : 10,   (string) The average receive bandwidth in Kbytes/sec\n"
+            "  }\n"
+            "\n NOTE: if the send and/or recv parameters do not exist, shaping in that direction is disabled.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("gettrafficshaping", "")
+            + HelpExampleRpc("gettrafficshaping", "")
+			    );
+    Object ret;
+    uint64_t max, ave;
+    sendShaper.get(&max,&ave);
+    if (ave != LONG_MAX)
+    {
+        ret.push_back(Pair("sendBurst", max/1024));
+        ret.push_back(Pair("sendAve", ave/1024));
+    }
+    receiveShaper.get(&max,&ave);
+    if (ave != LONG_MAX)
+    {
+        ret.push_back(Pair("recvBurst", max/1024));
+        ret.push_back(Pair("recvAve", ave/1024));
+    }
+    return ret;
+}
+
+Value settrafficshaping(const Array& params, bool fHelp)
+{
+    bool disable = false;
+    bool badArg = false;
+    string strCommand;
+    CLeakyBucket* bucket = NULL;
+    if (params.size() >= 2)
+      {
+        strCommand = params[0].get_str();
+        if (strCommand=="send") bucket = &sendShaper;
+        if (strCommand=="receive") bucket = &receiveShaper;
+        if (strCommand=="recv") bucket = &receiveShaper;
+      }
+    if (params.size() == 2)
+      {
+        if (params[1].get_str() == "disable") disable=true;
+        else badArg = true;
+      }
+    else if (params.size() != 3) badArg = true;
+
+    if (fHelp || badArg || bucket==NULL)        
+        throw runtime_error(
+            "settrafficshaping \"send|receive\" \"burstKB\" \"averageKB\""
+            "\nSets the network send or receive bandwidth and burst in kilobytes per second.\n"
+            "\nArguments:\n"
+            "1. \"send|receive\"     (string, required) Are you setting the transmit or receive bandwidth\n"
+            "2. \"burst\"  (integer, required) Specify the maximum burst size in Kbytes/sec (actual max will be 1 packet larger than this number)\n"
```

Comment:
I wonder if kiloBYTES/sec is really the right metric to specify here, given that bandwidth is normally measured in megaBITS/sec. Otherwise I guess nearly all real users will have to do the multiplication themselves. Seems a bit user unfriendly...